packages/inferno-animation/src/utils.ts

```diff
@@ -213,7 +213,7 @@ export function registerTransitionListener(nodes: HTMLElement[], callback: Funct
 
   // Fallback if transitionend fails
   // This is disabled during debug so we can set breakpoints
-  if (!window.__DEBUG_ANIMATIONS__ && !noTimeout) {
+  if (!isDebugAnimationsSet() && !noTimeout) {
```

Comment:
I meant moving the contents of this function outside, registerTransitionListener  otherwise we create 2 additional function scopes even if noTimeout or debug is false.

Also process.env.NODE_ENV !== 'production' check needs to be inlined here, because its special compiler time check, which will be replaced with true or false depending on the build target, then it will be fully removed from the output as a result

Reply:
Now I have checked that the actual code is removed in minified production build (i.e. the string '__INFERNO_ANIMATION_DEBUG__' doesn't exist). I also changed the name of the variable you set to window.__INFERNO_ANIMATION_DEBUG__.

In dev builds you can set `window.__INFERNO_ANIMATION_DEBUG__ = true` in the console to start debugging if an animation isn't behaving as expected (usually meaning you have forgotten to include the CSS, but could be caused by other issues if you have created your own animation component from scratch).

FIXED in 4366159fe